src/platform/datapath_epoll.c

```diff
@@ -2450,12 +2450,34 @@ CxPlatSendDataSend(
             Status = QUIC_STATUS_PENDING;
         } else {
             Status = errno;
+#ifdef UDP_SEGMENT
+            QuicTraceEvent(
+                DatapathErrorStatus,
+                "[data][%p] ERROR, %u, %s.",
+                SocketContext->Binding,
+                Status,
+                "sendmsg (GSO) failed");
+#else
             QuicTraceEvent(
                 DatapathErrorStatus,
                 "[data][%p] ERROR, %u, %s.",
                 SocketContext->Binding,
                 Status,
                 "sendmmsg failed");
+#endif
+
+            if (Status == EIO) {
```

Comment:
It may be worth also checking that the feature was used on the syscall and/or is still enabled. Unlikely to see EIO for other reasons, but will be confusing if so.